### DISCUSSION TITLE
Insert data from message header (or key) to clickhouse

### DISCUSSION THREAD

**Discussion by ne1r0n** [ClickHouse/clickhouse-kafka-connect#198] [Ideas]
I have a case where kafka topic contains some useful information inside message **header** and I would like to ingest that data to ClickHouse together with value.

I've tried to use [HeaderToField](https://jcustenborder.github.io/kafka-connect-documentation/projects/kafka-connect-transform-common/transformations/HeaderToField.html#headertofield) and [HeaderToValue](https://debezium.io/documentation/reference/stable/transformations/header-to-value.html) transforms, but it doesn't work with json w/o schemas.

I would be great if ClickHouse kafka connect could read and insert data from header and key into ClickHouse.
For the header to put a json with header's key/value into a one column would be more than enough or it could be a mapping in connect config to get exact key and put it to desired column.

**Comment by Paultagoras** [comment]
I have an idea for this @ne1r0n - let me take a stab at it and let you know...

We've already created a Single Message Transform for something else - it might make sense to use that here.

**Reply by ne1r0n** [reply]
The problem is 2 mentioned above SMTs don't work with schema-less json.
